### PR TITLE
RST-3818 - Fixed issue with form collection proxy when removing items

### PR DESCRIPTION
**Pull request by garytaylor**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by Fred83200 — APPROVED** [review]

File: app/javascript/packs/components/RemoveMultiple.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ function setupClickHandler() {
 
     e.preventDefault();
     removeMultiple(e.target.attributes['data-multiple-remove'].value)
+    showButton()
+    removePTag()
```
Comment (garytaylor) [inline app/javascript/packs/components/RemoveMultiple.js:19]: This needs a better name !  'Tag' means nothing :-)

File: app/javascript/packs/components/RemoveMultiple.js [hunk 2]
```diff
@@ -41,6 +43,16 @@ function removeMultiple(id) {
   updateCount();
 }
 
+function showButton() {
+  const button = document.querySelector('button[data-multiple-add]')
+  button.classList.remove('hidden')
+}
+
+function removePTag() {
+  const p = document.getElementById('multiple-respondent-p')
```
Comment (garytaylor) [inline app/javascript/packs/components/RemoveMultiple.js:52]: There is a much easier way - mark anything in the html with data: { 'multiple-add': true } that needs hiding and showing.  Add the hidden class in the html where needed

Then, change the showButton function and name it showHiddenElements - then change the css selector to be '*[data-multiple-add]'. - you then won't need the removePTag function

File: app/javascript/packs/components/RemoveMultiple.js (hunk 1 above)
Comment (garytaylor) [reply app/javascript/packs/components/RemoveMultiple.js:19]: but see my comment below first - you might not need this function

File: app/javascript/packs/components/RemoveMultiple.js (hunk 2 above)
Comment (garytaylor) [reply app/javascript/packs/components/RemoveMultiple.js:52]: You will also need to change the document.querySelectorAll and loop over its results and remove the hidden class from all elements found

File: app/javascript/packs/components/RemoveMultiple.js [hunk 3]
```diff
@@ -15,6 +15,8 @@ function setupClickHandler() {
 
     e.preventDefault();
     removeMultiple(e.target.attributes['data-multiple-remove'].value)
+    showButton()
+    removeMultipleRespondentPTag()
```
Comment (garytaylor) [inline app/javascript/packs/components/RemoveMultiple.js:19]: I know it doesn't seem important - but take another look at the name of this function.  Imagine you are someone new coming to the code base and you are looking at the page in your browser.  Does the name 'removeMultipleRespondetPTag' mean anything to you ?  it doesn't to me.

My guess is that you want to name it based on what the content you are removing actually is ?

Comment (garytaylor) [reply app/javascript/packs/components/RemoveMultiple.js:19]: Also, RemoveMultiple.js is not specific to multiple respondents - I believe it is used for both respondents and claimants - so unsure this is even the right place for it ?

**Pull request merged**